lib/OptionsDefaulter.js

```diff
@@ -27,44 +23,53 @@ function setProperty(obj, name, value) {
 	obj[name.pop()] = value;
 }
 
-OptionsDefaulter.prototype.process = function(options) {
-	var addItemTo = function addItemTo(list) {
-		return function(item) {
-			list.push(item);
-		}
+class OptionsDefaulter {
+	constructor() {
+		this.defaults = {}
+		this.config = {};
 	}
 
-	for(var name in this.defaults) {
-		switch(this.config[name]) {
-			case undefined:
-				if(getProperty(options, name) === undefined)
-					setProperty(options, name, this.defaults[name]);
-				break;
-			case "call":
-				setProperty(options, name, this.defaults[name].call(this, getProperty(options, name), options), options);
-				break;
-			case "make":
-				if(getProperty(options, name) === undefined)
-					setProperty(options, name, this.defaults[name].call(this, options), options);
-				break;
-			case "append":
-				var oldValue = getProperty(options, name);
-				if(!Array.isArray(oldValue)) oldValue = [];
-				this.defaults[name].forEach(addItemTo(oldValue));
-				setProperty(options, name, oldValue);
-				break;
-			default:
-				throw new Error("OptionsDefaulter cannot process " + this.config[name]);
+	process(options) {
+		const addItemTo = function addItemTo(list) {
```

Comment:
Also, note, that writing this as arrow functions only loses the `Function#name` because Node 4's implementation is incomplete. The outermost arrow function is supposed to receive its contextual name from the `addItemTo =` assignment.